cli/contract.go

```diff
@@ -0,0 +1,145 @@
+package cli
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+	"go.keploy.io/server/v2/config"
+	contractSvc "go.keploy.io/server/v2/pkg/service/contract"
+	"go.keploy.io/server/v2/utils"
+	"go.uber.org/zap"
+)
+
+func init() {
+	Register("contract", Contract)
+}
+
+func Contract(ctx context.Context, logger *zap.Logger, _ *config.Config, serviceFactory ServiceFactory, cmdConfigurator CmdConfigurator) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "contract",
+		Short: "Manage keploy contracts",
+	}
+
+	cmd.AddCommand(Generate(ctx, logger, serviceFactory, cmdConfigurator, cmd.Name()))
+	cmd.AddCommand(Download(ctx, logger, serviceFactory, cmdConfigurator, cmd.Name()))
+	cmd.AddCommand(Validate(ctx, logger, serviceFactory, cmdConfigurator, cmd.Name()))
+
+	return cmd
+}
+
+func Generate(ctx context.Context, logger *zap.Logger, serviceFactory ServiceFactory, cmdConfigurator CmdConfigurator, parentCmd string) *cobra.Command {
```

Comment:
Rather than passing the parent command, I guess you can use: `cmd.Parent().Name()` Please check once.

Reply:
I'm passing the name already , **parentCmd** is a string variable refers to the name of the parent cmd.

Reply:
Yes, but you should not pass if it can be get without sending the parent command.